osu.Game/Overlays/Profile/Sections/Beatmaps/PaginatedBeatmapContainer.cs

```diff
@@ -49,15 +57,89 @@ protected override void ShowMore()
                     var panel = new DirectGridPanel(s.ToBeatmapSet(Rulesets));
                     ItemsContainer.Add(panel);
                 }
+
+                MoreButton.FadeTo(ItemsContainer.Children.Count == count ? 0 : 1);
             });
 
             Api.Queue(request);
         }
 
+        protected override void OnUserChanged(ValueChangedEvent<User> e)
+        {
+            base.OnUserChanged(e);
+
+            switch (type)
+            {
+                case BeatmapSetType.Favourite:
+                    count = User.Value.FavouriteBeatmapsetCount[0];
+                    break;
+
+                case BeatmapSetType.Graveyard:
+                    count = User.Value.GraveyardBeatmapsetCount[0];
+                    break;
+
+                case BeatmapSetType.Loved:
+                    count = User.Value.LovedBeatmapsetCount[0];
+                    break;
+
+                case BeatmapSetType.RankedAndApproved:
+                    count = User.Value.RankedAndApprovedBeatmapsetCount[0];
+                    break;
+
+                case BeatmapSetType.Unranked:
+                    count = User.Value.UnrankedBeatmapsetCount[0];
+                    break;
+            }
+
+            counterDrawable.Value = count;
+            counterDrawable.FadeTo(count > 0 ? 1 : 0);
```

Comment:
The fading should be moved inside `Counter`, also, i suggest using a bindable and bind it to `Counter` instead of 2 fields